fl4health/strategies/feddg_ga_strategy.py

```diff
@@ -0,0 +1,382 @@
+from enum import Enum
+from logging import WARNING
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+from flwr.common import MetricsAggregationFn, NDArrays, Parameters, ndarrays_to_parameters, parameters_to_ndarrays
+from flwr.common.logger import log
+from flwr.common.typing import EvaluateRes, FitIns, FitRes, Scalar
+from flwr.server.client_manager import ClientManager
+from flwr.server.client_proxy import ClientProxy
+from flwr.server.strategy import FedAvg
+
+from fl4health.client_managers.fixed_sampling_client_manager import FixedSamplingClientManager
+
+
+class FairnessMetricType(Enum):
+    """Defines the basic types for fairness metrics, their default names and their default signals"""
+
+    ACCURACY = "val - prediction - accuracy"
+    LOSS = "val - loss"
+    CUSTOM = "custom"
+
+    @classmethod
+    def signal_for_type(cls, fairness_metric_type: "FairnessMetricType") -> float:
+        """
+        Return the default signal for the given metric type.
+
+        Args:
+            fairness_metric_type: (FairnessMetricType) the fairness metric type.
+
+        Returns: (float) -1.0 if FairnessMetricType.ACCURACY, 1.0 if FairnessMetricType.LOSS and
+            0 if FairnessMetricType.CUSTOM.
+        """
+        if fairness_metric_type == FairnessMetricType.ACCURACY:
+            return -1.0
+        if fairness_metric_type == FairnessMetricType.LOSS:
+            return 1
+        return 0
```

Comment:
I think returning a 0 here is sort of "hiding" an issue. If I understand how the signal value works below, a signal value of 0 essentially means that we're nulling any updates to the GA weights. Perhaps we can through an error (or warning) here instead stating the resulting signal will be zero, which is very likely not what the user wants.

Reply:
Yep, makes sense. Will change to raise an exception.